lib/xnvme_be_macos_driverkit_sync.c

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: Samsung Electronics Co., Ltd
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <libxnvme.h>
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_MACOS_ENABLED
+#include <xnvme_be_macos_driverkit.h>
+#include <xnvme_dev.h>
+
+static int
+cmd_io(struct xnvme_cmd_ctx *ctx, void *dbuf, size_t dbuf_nbytes, void *mbuf, size_t mbuf_nbytes)
+{
+	NvmeSubmitCmd cmd = {0};
+	NvmeSubmitCmd cmd_return = {0};
+	uint64_t _buf_base_offset;
```

Comment:
See notes on declaration / initialization order and error-handling.

Reply:
Done.